### PR TITLE
Add Makefile as management tool

### LINKED ISSUE #3: Use make for project management ?
I wanted to know whether we wanted to add a Makefile to handle all basic project management tasks, such as formatting, testing, linting ?

After building this make utility we could use it in our CI to keep the project clean and well tested.

### PR DESCRIPTION
**Pull request by jooola**
This adds a 'kinda' generic makefile for python project management with linting and formatting.

- Make the use of python3 venv module
- Formatting using Black
- Linting using Pylint
- Supercharge gitignore

Fixes #3

### CODE REVIEW

**Comment by jooola** [comment]
Fixes #3

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+black
+pylint
```
Comment (vvatelot) [inline requirements.txt]: Maybe in `requirements.dev.txt`?

Comment (jooola) [reply requirements.txt]: Yup good enough for me.

**Comment by vvatelot** [comment]
@jooola what about a small how to start for dev in the readme?

**Comment by vvatelot** [comment]
@jooola made a PR on this branch to update README file

**Pull request merged**